### PR TITLE
add gotify notification client

### PR DESCRIPTION
**Pull request by pagdot**
Add support to push notifications to gotify.

### CODE REVIEW

**Comment by pagdot** [comment]
docker build timed out after 6h on the server while fetching base images.

I've tested my changes with my gotify instance an I am able to receive push messages

**Comment by crazy-max** [comment]
Hi @pagdot, thanks your contrib!

> docker build timed out after 6h on the server while fetching base images.

Looks like a timeout on the DockerHub side, no big deal. Now it's ok. 

> I've tested my changes with my gotify instance an I am able to receive push messages

I'll do a review asap.

**Comment by pagdot** [comment]
Yeah, they even offer the go command to activate the webhook.

File: internal/notif/gotify/client.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		title = fmt.Sprintf("New image %s has been added", entry.Image.String())
+	}
+
+	var msgBuf bytes.Buffer
+	msgTpl := template.Must(template.New("email").Parse(`Docker 🐳 tag {{ .Image.Domain }}/{{ .Image.Path }}:{{ .Image.Tag }} which you subscribed to through {{ .Provider }} provider has been {{ if (eq .Status "new") }}newly added{{ else }}updated{{ end }}.`))
+	if err := msgTpl.Execute(&msgBuf, entry); err != nil {
+		return err
+	}
+
+	_, err := hc.PostForm(c.cfg.Host+"/message?token="+c.cfg.Token,
```
Comment (crazy-max) [inline internal/notif/gotify/client.go]: Would be better to sanitize `c.cfg.Host` with [`url.Parse()`](https://golang.org/pkg/net/url/#Parse) and use `URL` object to set path and query value.

File: internal/notif/gotify/client.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		url.Values{"message": {msgBuf.String()}, "title": {title}})
```
Comment (crazy-max) [inline internal/notif/gotify/client.go]: Use a proper struct as defined in https://gotify.net/api-docs#/message/createMessage. Also add `priority` maybe?

File: internal/notif/gotify/client.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+	return err
```
Comment (crazy-max) [inline internal/notif/gotify/client.go]: Also, handle HTTP response to return `errorDescription` as defined in https://gotify.net/api-docs#/message/createMessage

File: internal/notif/gotify/client.go (hunk 2 above)
Comment (pagdot) [reply internal/notif/gotify/client.go]: I've pretty much used this sample for go: https://gotify.net/docs/more-pushmsg

**Comment by pagdot** [comment]
Yeah, sorry. These are my first lines ~writing~ copy-pasting go. I'm okay with either of us making the changes

**Comment by crazy-max** [comment]
@pagdot That's fine, I take care of it ;)

**Pull request closed without merging**